engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageState.java

```diff
@@ -18,14 +18,18 @@
 import io.camunda.zeebe.db.impl.DbLong;
 import io.camunda.zeebe.db.impl.DbNil;
 import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.db.impl.rocksdb.BufferedMessagesMetrics;
 import io.camunda.zeebe.engine.state.mutable.MutableMessageState;
 import io.camunda.zeebe.protocol.ZbColumnFamilies;
 import io.camunda.zeebe.protocol.impl.record.value.message.MessageRecord;
+import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
 import org.agrona.DirectBuffer;
 import org.agrona.collections.MutableBoolean;
 
 public final class DbMessageState implements MutableMessageState {
 
+  private static final Long DEADLINE_MESSAGE_COUNT_KEY = 1L;
```

Comment:
:wrench: IMO this is a bit weird to be 1 as the key.  I would recommend to have a more general stats column family and specific keys.

```
private static final String DEADLINE_MESSAGE_COUNT_KEY = "deadline_message_count";
private final ColumnFamily<DbString, DbLong> messageStatsColumnFamily;

....
  messagesDeadlineCount = new DbLong();
  messagesDeadlineCountKey = new DbString();
  messageStatsColumnFamily =
      zeebeDb.createColumnFamily(
  ZbColumnFamilies.MESSAGE_STATS,
  transactionContext,
  messagesDeadlineCountKey,
  messagesDeadlineCount);

  messagesDeadlineCountKey.wrapString(DEADLINE_MESSAGE_COUNT_KEY);
```

Or have a generic column family for such metrics, so that we can add more metrics in future.

```
  statsColumnFamiliy =
      zeebeDb.createColumnFamily(
  ZbColumnFamilies.STATS_COUNT,
  transactionContext,
  messagesDeadlineCountKey,
  messagesDeadlineCount);
```

What do you think?

Reply:
I think having a general MESSAGE_STATS is good as it can be any type of stat and not just count 🤔

Reply:
>  it can be any type of stat and not just count

Remember that the type of the value in a columnfamily is pre-defined. So you cannot store anything in a general MESSAGE_STATS, but only numbers.